### PR TITLE
Add postgres acceptance tests

### PR DESCRIPTION
**Pull request by adfoster-r7**
Add postgres acceptance tests

### Github view

image

### Allure report generation

image

### Local test module

Creating session

```
msf6 auxiliary(scanner/postgres/postgres_login) > run rhost=127.0.0.1 username=postgres password=password rport=9000 createsession=true

[!] No active DB -- Credential data will not be saved!
[+] 127.0.0.1:9000 - Login Successful: postgres:password@template1
[*] PostgreSQL session 1 opened (127.0.0.1:50172 -> 127.0.0.1:9000) at 2024-03-07 10:05:21 +0000
...
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
smsf6 auxiliary(scanner/postgres/postgres_login) > 
sessionsmsf6 auxiliary(scanner/postgres/postgres_login) > sessions

Active sessions
===============

  Id  Name  Type        Information                           Connection
  --  ----  ----        -----------                           ----------
  1         postgresql  PostgreSQL postgres @ 127.0.0.1:9000  127.0.0.1:50172 -> 127.0.0.1:9000 (127.0.0.1)
```

Loading test module

```
msf6 auxiliary(scanner/postgres/postgres_login) > loadpath test/modules
Loaded 40 modules:
    14 auxiliary modules
    13 exploit modules
    13 post modules
```

Running test module

```
msf6 auxiliary(scanner/postgres/postgres_login) > use post/test/postgres
msf6 post(test/postgres) > run session=-1

[!] SESSION may not be compatible with this module:
[!]  * Unknown session platform. This module works with: Apple_iOS, Hardware, Multi, Mainframe, Firefox, NodeJS, Python, JavaScript, PHP, Unix, Irix, HPUX, AIX, FreeBSD, NetBSD, BSDi, OpenBSD, BSD, OSX, Solaris, Arista, Mikrotik, Brocade, Unifi, Juniper, Cisco, Linux, Ruby, R, Java, Android, Netware, Windows, Unknown.
[*] Running against session -1
[*] Session type is postgresql and platform is 
[+] should return a version
[+] should support the help command
[*] Testing complete in 1.75 seconds
[*] Passed: 2; Failed: 0; Skipped: 0
[*] Post module execution completed
msf6 post(test/postgres) > 
```

## Verification

- Ensure CI passes
- Ensure local replication docs work

### CODE REVIEW

**Comment by cgranleese-r7** [comment]
Tested locally and got the expected output:
image

File: .github/workflows/acceptance.yml
Comment (adfoster-r7) [inline .github/workflows/acceptance.yml:1]: I'm keeping the meterpreter acceptance testing separate for now; but in the future I'd like to merge the upcoming postgres/smb/mssql/etc tests into the same allure report 🍏

**Comment by cgranleese-r7** [comment]
Retested after the changes and everything worked as expected 👍 
image

**Comment by cgranleese-r7** [comment]
## Release Notes
This PR adds PostgreSQL session type acceptance tests, with Allure report generation as well as a local test module.

**Pull request merged**